presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/GroupInnerJoinsByConnectorRuleSet.java

```diff
@@ -203,9 +207,9 @@ public static class FilterOnJoinRule
     {
         private static final Capture<JoinNode> JOIN = newCapture();
 
-        public FilterOnJoinRule(Metadata metadata)
+        public FilterOnJoinRule(Metadata metadata, PlanOptimizer predicatePushdownOptimizer)
```

Comment:
@Haritha-Koloth Can you share a query where we see a `FilterNode<-JoinNode` at this point in the planner ? I'm wondering if this rule variant is actually needed